platform/mellanox/mlnx-platform-api/sonic_platform/dpuctl_hwm.py

```diff
@@ -0,0 +1,163 @@
+# limitations under the License.
+#
+
+"""Class Implementation for per DPU functionality"""
+import os.path
+import time
+
+try:
+    from .inotify_helper import InotifyHelper
+    from sonic_py_common.logger import Logger
+    from sonic_platform import utils
+except ImportError as e:
+    raise ImportError(str(e)) from e
+
+EVENT_BASE = "/var/run/hw-management/events/"
+SYSTEM_BASE = "/var/run/hw-management/system/"
+CONFIG_BASE = "/var/run/hw-management/config"
+logger = Logger()
+
+WAIT_FOR_SHTDN = 60
+WAIT_FOR_DPU_READY = 60
+
+
+class DataWriter():
+    """Class for writing data to files"""
+    def __init__(self, file_name):
+        self.file_name = file_name
+        self.file_obj = None
+        if not os.path.isfile(self.file_name):
+            raise FileNotFoundError(f"File {self.file_name} does not exist!")
+
+    def __enter__(self):
+        self.file_obj = open(self.file_name, 'w', encoding="utf-8")
+        return self.file_obj
+
+    def __exit__(self, *args):
+        self.file_obj.close()
+
+
+class DpuCtlPlat():
+    """Class for Per DPU API Call"""
+    def __init__(self, dpu_name):
+        self._name = dpu_name
+        self.set_go_down_path = os.path.join(SYSTEM_BASE,
+                                             f"{self._name}_rst")
+        self.set_pwr_path = os.path.join(SYSTEM_BASE,
+                                         f"{self._name}_pwr")
+        self.set_pwr_f_path = os.path.join(SYSTEM_BASE,
+                                           f"{self._name}_pwr_force")
+        self.get_dpu_rdy_path = os.path.join(EVENT_BASE,
+                                             f"{self._name}_ready")
+        self.set_dpu_perst_en_path = os.path.join(SYSTEM_BASE,
+                                                  f"{self._name}_perst_en")
+
+    def write_file(self, file_name, content_towrite):
+        """Write given value to file only if file exists"""
+        try:
+            with DataWriter(file_name) as file_obj:
+                file_obj.write(content_towrite)
+        except Exception as file_write_exc:
+            logger.log_error(f'{self.get_name()}:Failed to write'
+                             f'{content_towrite} to file {file_name}')
+            raise type(file_write_exc)(
+                f"{self.get_name()}:{str(file_write_exc)}")
+        return True
+
+    def get_name(self):
+        """Return name of the DPU"""
+        return self._name
+
+    def dpu_go_down(self):
+        """Per DPU going down API"""
+        get_shtdn_ready_path = os.path.join(EVENT_BASE,
+                                            f"{self.get_name()}_shtdn_ready")
+        try:
+            get_shtdn_inotify = InotifyHelper(get_shtdn_ready_path)
+            dpu_shtdn_rdy = get_shtdn_inotify.add_watch(WAIT_FOR_SHTDN, 1)
+        except (FileNotFoundError, PermissionError) as inotify_exc:
+            raise type(inotify_exc)(f"{self.get_name()}:{str(inotify_exc)}")
+        if dpu_shtdn_rdy is None:
```

Comment:
if not dpu_shtdn_rdy?

Reply:
`add_watch` returns value which is read from the file (if the value is matching the expected value and returns `None` if the values do not match, so checking for `None` specifically)